server/utils/passport/ensure-authenticated.js

```diff
@@ -1,33 +1,31 @@
-import basePathConfig from '../../configs/basePathConfig'
+import UserModel from '../../models/UserModel'
+import { isAccountExpired } from './helpers'
 
-const basePath = basePathConfig || ''
-const logoutForbiddenRoute = `${basePath}/logout?e=forbidden`
-const unauthorizedRoute = `${basePath}/logout?e=unauthorized`
+export default async function ensureAuthenticated(req, res, next) {
+  try {
+    if (!req.isAuthenticated())
+      return res.status(401).json({ error: 'Please login' })
 
-export default function ensureAuthenticated(req, res, next) {
-  if (!req.isAuthenticated()) {
-    return res.redirect(`${basePath}/logout`)
+    const { appDb } = req.app.locals
+    const uid = req.user
+    const user = await UserModel.findOne(appDb, uid)
+    const { account_expires, blocked, access } = user
+
+    switch (true) {
+      case isAccountExpired(account_expires):
+        return res.status(401).json({ error: 'Account is expired' })
+      case !!blocked:
+        return res
+          .status(403)
+          .json({ error: 'Account is blocked, please contact your admin' })
+      case access.length === 0:
+        return res.status(403).json({ error: 'Forbidden.' })
+      default:
+        req.app.locals.currentUser = user
```

Comment:
do we want to store the user here? If so we'll need to keep the user up to date if they make changes like editing their profile. Might be safer to store the user's ID (which I think we already have in `req.user`) and then get the user from the DB.

Reply:
[a6af4fd](https://github.com/AMP-SCZ/dpdash/pull/317/commits/a6af4fdc187f5764b6233074f05fd8ff2993e74a)